src/scirpy/tests/test_ir_dist_metrics.py

```diff
@@ -306,11 +307,327 @@ def test_fast_alignment_dist_with_two_seq_arrays():
+        (
+            {
+                "dist_weight": 3,
+                "gap_penalty": 4,
+                "ntrim": 3,
+                "ctrim": 2,
+                "fixed_gappos": False,
+                "cutoff": 1000,
+                "n_jobs": 1,
+            },
+            (
+                np.array(["AAAAAAAAAAAAAAA", "AAAAAAAAARRAAAA", "AAAAAAANDAAAA"]),
+                np.array(["AAAAAAAAAAAAAAA", "AAAAAAAAARRAAAA", "AAAAAAANDAAAA"]),
+            ),
+            np.array([[1, 25, 33], [25, 1, 33], [33, 33, 1]]),
+        ),
+        # test with cutoff set to 0
+        (
+            {
+                "dist_weight": 3,
+                "gap_penalty": 4,
+                "ntrim": 3,
+                "ctrim": 2,
+                "fixed_gappos": True,
+                "cutoff": 0,
+                "n_jobs": 1,
+            },
+            (
+                np.array(["AAAAAAAAAA", "AAAARRAAAA", "AANDAAAA"]),
+                np.array(["AAAAAAAAAA", "AAAARRAAAA", "AANDAAAA"]),
+            ),
+            np.array([[1, 0, 0], [0, 1, 0], [0, 0, 1]]),
+        ),
+        # test with cutoff set high to neglect it
+        (
+            {
+                "dist_weight": 3,
+                "gap_penalty": 4,
+                "ntrim": 3,
+                "ctrim": 2,
+                "fixed_gappos": True,
+                "cutoff": 1000,
+                "n_jobs": 1,
+            },
+            (
+                np.array(["AAAAAAAAAA", "AAAARRAAAA", "AANDAAAA"]),
+                np.array(["AAAAAAAAAA", "AAAARRAAAA", "AANDAAAA"]),
+            ),
+            np.array([[1, 25, 21], [25, 1, 21], [21, 21, 1]]),
+        ),
+        # test with multiple cores by setting n_jobs = 4
+        (
+            {
+                "dist_weight": 3,
+                "gap_penalty": 4,
+                "ntrim": 3,
+                "ctrim": 2,
+                "fixed_gappos": True,
+                "cutoff": 20,
+                "n_jobs": 4,
+            },
+            (
+                np.array(["AAAAAAAAAA", "AAAARRAAAA", "AANDAAAA"]),
+                np.array(["AAAAAAAAAA", "AAAARRAAAA", "AANDAAAA"]),
+            ),
+            np.array([[1, 0, 21], [0, 1, 21], [21, 21, 1]]),
+        ),
+    ],
+)
+def test_tcrdist(test_parameters, test_input, expected_result):
+    tcrdist_calculator = TCRdistDistanceCalculator(**test_parameters)
+    seq1, seq2 = test_input
+    res = tcrdist_calculator.calc_dist_mat(seq1, seq2)
+    assert isinstance(res, scipy.sparse.csr_matrix)
+    assert res.shape == expected_result.shape
+    assert np.array_equal(res.todense(), expected_result)
+
+
+def test_tcrdist_reference():
+    # test tcrdist against reference implementation
+    seqs = np.load("data/tcrdist_test_data/tcrdist_WU3k_seqs.npy")
+    reference_result = scipy.sparse.load_npz("data/tcrdist_test_data/tcrdist_WU3k_csr_result.npz")
```

Comment:
This ensures that the test data folder is always found, independent of what the developer sets as root of the test directory. 

```suggestion
    from . import TESTDATA
    seqs = np.load(TESTDATA / "tcrdist_test_data/tcrdist_WU3k_seqs.npy")
    reference_result = scipy.sparse.load_npz(TESTDATA / "tcrdist_test_data/tcrdist_WU3k_csr_result.npz")
```